yarn-project/circuits.js/src/contract/contract_class.ts

```diff
@@ -22,6 +22,8 @@ export function getContractClassFromArtifact(
   const artifactPublicFunctions: ContractClass['publicFunctions'] = artifact.functions
     .filter(f => f.functionType === FunctionType.PUBLIC)
     .map(f => ({
+      // TODO: #10752 consider removing function name from contract class
+      name: f.name,
```

Comment:
Heads up this never makes it to the actual contract class that gets published to the node. Since the change so that we have only one public bytecode for the entire contract, we only publish the "dispatch function" (see below in this same file) which has the combined bytecode of all public functions, so this name gets lost.

Reply:
oh right, I should've read the whole function not stop at the first sign of public functions